File: app/src/main/java/com/mdgd/pokemon/PokemonsApp.java
package com.mdgd.pokemon;

import androidx.multidex.MultiDexApplication;

import com.mdgd.pokemon.models.AppModule;
import com.mdgd.pokemon.models_impl.DefaultAppModule;

public class PokemonsApp extends MultiDexApplication {

    private static PokemonsApp instance;
    private AppModule appComponent;

    public static PokemonsApp getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        appComponent = new DefaultAppModule(this);
    }

    public AppModule getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/com/mdgd/pokemon/PokemonsApp.kt
package com.mdgd.pokemon

import androidx.hilt.work.HiltWorkerFactory
import androidx.multidex.MultiDexApplication
import androidx.work.Configuration
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

@HiltAndroidApp
class PokemonsApp : MultiDexApplication(), Configuration.Provider {

    companion object {
        var instance: PokemonsApp? = null
            private set
    }

    @Inject
    lateinit var workerFactory: HiltWorkerFactory

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    override fun getWorkManagerConfiguration() = Configuration.Builder()
        .setWorkerFactory(workerFactory)
        .build()
}
